File: Loenn/entities/judgmentBell.lua
local utils = require("utils")
local drawableSprite = require("structs.drawable_sprite")

local judgmentBell = {}

judgmentBell.name = "MaggyHelper/JudgmentBell"
judgmentBell.depth = -100
judgmentBell.fieldInformation = {
    shockwaveSpeed = {
        minimumValue = 0
    },
    shockwaveRadius = {
        minimumValue = 0
    },
    cooldownTime = {
        minimumValue = 0
    },
    maxRings = {
        minimumValue = 1
    }
}
judgmentBell.placements = {
    {
        name = "normal",
        data = {
            shockwaveSpeed = 200,
            shockwaveRadius = 300,
            cooldownTime = 2.0,
            maxRings = 3,
            canKglobal::PlayerRing = true
        }
    }
}

function judgmentBell.sprite(room, entity)
    local sprite = drawableSprite.fromTexture("objects/judgment_bell/idle", entity)
    sprite:setPosition(entity.x, entity.y)
    sprite:setJustification(0.5, 1.0)
    sprite.color = {0.8, 0.7, 0.4, 1.0}
    return sprite
end

function judgmentBell.selection(room, entity)
    return utils.rectangle(entity.x - 24, entity.y - 48, 48, 48)
end

return judgmentBell

File: framework/extracted/Ahorn/entities/JudgmentBell.jl
module JudgmentBell

using ..Ahorn, Maple

@mapdef Entity "MaggyHelper/JudgmentBell" JudgmentBell(x::Integer, y::Integer, PlayerRing::Bool=true, cooldownTime::Number=2.0, maxRings::Integer=3, shockwaveRadius::Integer=300, shockwaveSpeed::Integer=200)

const placements = Ahorn.PlacementDict(
    "normal" => Ahorn.EntityPlacement(JudgmentBell)
)

function Ahorn.selection(entity::JudgmentBell)
    return Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::JudgmentBell, room::Maple.Room)
    Ahorn.drawRectangle(ctx, Ahorn.Rectangle(entity.x - 8, entity.y - 8, 16, 16), (0.5, 0.5, 0.5, 0.5), (0.0, 0.0, 0.0, 1.0))
end

end
